oval-schemas/independent-definitions-schema.xsd

```diff
@@ -968,6 +968,171 @@
+                  </xsd:complexContent>
+            </xsd:complexType>
+      </xsd:element>
+      
+      <xsd:element name="shellcommand_object" substitutionGroup="oval-def:object">
+            <xsd:annotation>
+                  <xsd:documentation>The shellcommand_object is used by a shellcommand_test to define a shell to use (e.g. sh, bash, ksh, etc.), a command (or shell script) to be run, and a pattern to filter result lines. The default shell is bash.  Each object extends the standard ObjectType as defined in the oval-definitions-schema and one should refer to the ObjectType description for more information. The common set element allows complex objects to be created using filters and set logic.
+                  
+The evaluation of the object should always produce at least one item.  If the object evaluation does not produce output that should result in an item, one should be created with a status of 'does not exist' and, if possible, the exit_status and any STDERR from the system call should be captured in that item.
+                  </xsd:documentation>
+                  <xsd:appinfo>
+                        <sch:pattern id="ind-def_shellcommand_object_verify_filter_state">
+                              <sch:rule context="ind-def:shellcommand_object//oval-def:filter">
+                                    <sch:let name="parent_object" value="ancestor::ind-def:shellcommand_object"/>
+                                    <sch:let name="parent_object_id" value="$parent_object/@id"/>
+                                    <sch:let name="state_ref" value="."/>
+                                    <sch:let name="reffed_state" value="ancestor::oval-def:oval_definitions/oval-def:states/*[@id=$state_ref]"/>
+                                    <sch:let name="state_name" value="local-name($reffed_state)"/>
+                                    <sch:let name="state_namespace" value="namespace-uri($reffed_state)"/>
+                                    <sch:assert test="(($state_namespace='http://oval.mitre.org/XMLSchema/oval-definitions-5#independent') and ($state_name='shellcommand_state'))">State referenced in filter for <sch:value-of select="name($parent_object)"/> '<sch:value-of select="$parent_object_id"/>' is of the wrong type. </sch:assert>
+                              </sch:rule>
+                        </sch:pattern>
+                  </xsd:appinfo>
+            </xsd:annotation>
+            <xsd:complexType>
+                  <xsd:complexContent>
+                        <xsd:extension base="oval-def:ObjectType">
+                              <xsd:sequence>
+                                    <xsd:choice>
+                                          <xsd:element ref="oval-def:set"/>
+                                          <xsd:sequence>
+                                                <xsd:element name="shell" type="oval-def:EntityObjectStringType">
+                                                      <xsd:annotation>
```

Comment:
I think it's potentially problematic to use a relative path and an arbitrary string value for specifying the path to the shell.  How about an enumeration?

If we do an enumeration, I'd have values for: Korn, Bourne, csh, tcsh, bash, zsh, /usr/bin/env, cmd.exe, powershell.exe and "" (empty string to allow for a variable value).

Reply:
OOPS!  You already have an enumeration (EntityObjectShellType).  You just set the type attribute wrong for the shell entity.

Reply:
Yeah this was likely due to my updates and then John's, I'll fix the shell to use the enumeration.